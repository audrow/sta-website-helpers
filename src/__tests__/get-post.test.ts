import {
  // getPost,
  getNumber,
  getSlug,
  getMp3Url,
  getCustomMp3Url,
  // getIncludes,
} from '../get-post'

// describe('getPost', () => {
//   const episodesDirectory = join()
//   it('should get posts data', async () => {
//     //
//   })
// })

describe('getMp3Url', () => {
  const baseUrl = 'https://example.com'
  it('should return the expected URL', () => {
    expect(getMp3Url(baseUrl, 1, 'John Doe')).toBe(
      'https://example.com/STA Ep 1 - John Doe.mp3',
    )
    expect(getMp3Url(baseUrl, 1, ['John Doe'])).toBe(
      'https://example.com/STA Ep 1 - John Doe.mp3',
    )
    expect(getMp3Url(baseUrl, 1, ['John Doe', 'Bob Smith'])).toBe(
      'https://example.com/STA Ep 1 - John Doe and Bob Smith.mp3',
    )
    expect(getMp3Url(baseUrl, 1, ['John Doe', 'Bob Smith', 'Sara Wong'])).toBe(
      'https://example.com/STA Ep 1 - John Doe, Bob Smith, and Sara Wong.mp3',
    )
    expect(
      getMp3Url(baseUrl, 1, [
        'John Doe',
        'Bob Smith',
        'Sara Wong',
        'Margaret Brown',
      ]),
    ).toBe(
      'https://example.com/STA Ep 1 - John Doe, Bob Smith, Sara Wong, and Margaret Brown.mp3',
    )
  })
  it('should error when given an empty string', () => {
    expect(() => getMp3Url(baseUrl, 1, '')).toThrowError()
  })
  it('should error when given an empty list', () => {
    expect(() => getMp3Url(baseUrl, 1, [])).toThrowError(
      'Must have at least one guest',
    )
  })
})

describe('getCustomHostedMp3Url', () => {
  const baseUrl = 'https://example.com'
  it('should return the expected URL', () => {
    expect(getCustomMp3Url(baseUrl, 'Welcome')).toBe(
      'https://example.com/Welcome.mp3',
    )
  })
  it('should error when given an empty string', () => {
    expect(() => getCustomMp3Url(baseUrl, '')).toThrowError()
  })
})

describe('getSlug', () => {
  it('should return a slug', () => {
    expect(getSlug(1, ['foo'])).toEqual('1-foo')
    expect(getSlug(1, ['foo bar'])).toEqual('1-foo-bar')
    expect(getSlug(1, ['foo bar', 'baz bop'])).toEqual('1-foo-bar-and-baz-bop')
    expect(getSlug(1, ['foo bar', 'baz bop', 'bang pow'])).toEqual(
      '1-foo-bar-baz-bop-and-bang-pow',
    )
  })
  it('should throw an error on an empty list', () => {
    expect(() => getSlug(1, [])).toThrowError()
  })
  it('should throw errors on white space', () => {
    expect(() => getSlug(1, [''])).toThrowError()
    expect(() => getSlug(1, [' '])).toThrowError()
    expect(() => getSlug(1, ['   \n\t  '])).toThrowError()
    expect(() => getSlug(1, ['', ''])).toThrowError()
    expect(() => getSlug(1, ['', ' '])).toThrowError()
    expect(() => getSlug(1, ['foo', ''])).toThrowError()
    expect(() => getSlug(1, ['foo', ' '])).toThrowError()
    expect(() => getSlug(1, ['foo', '    \n\t   '])).toThrowError()
  })
})

describe('getPostNumberFromPath', () => {
  it('should return an episode number', () => {
    expect(getNumber('1')).toEqual(1)
    expect(getNumber('101')).toEqual(101)

    expect(getNumber('foo/1')).toEqual(1)

    expect(getNumber('foo/bar/1')).toEqual(1)
    expect(getNumber('foo/bar/2')).toEqual(2)
    expect(getNumber('foo/bar/101')).toEqual(101)
  })
  it('should return NaN if no number is given', () => {
    for (const path of ['', 'foo', 'foo/', 'foo/bar', 'foo/bar/']) {
      expect(getNumber(path)).toBeUndefined()
    }
  })
})
