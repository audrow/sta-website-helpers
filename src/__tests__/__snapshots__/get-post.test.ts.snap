// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPost should get a standard episode with a number 1`] = `
Object {
  "description": "In this episode, Audrow Nash speaks to Charles Brian Quinn (aka, CBQ), CEO and a Co-Founder of Greenzie. Greenzie make an autonomous driving system for commercial lawn mowers. We talk about Greenzie's autonomous mowing system, how Greenzie has worked with manufacturers to up-fit their system into comercial mowers, how Greenzie does dog-fooding, safety and standards, and about CBQ's experiences bootstrapping and with venture capital.",
  "duration": Object {
    "hours": 1,
    "minutes": 51,
    "seconds": 46,
  },
  "excerpt": "CBQ is the CEO and a Co-Founder of Greenzie.",
  "guests": "CBQ",
  "links": Array [
    Object {
      "name": "CBQ's website",
      "url": "https://www.seebq.com/",
    },
    Object {
      "name": "Greenzie's website",
      "url": "https://www.greenzie.com/",
    },
    Object {
      "name": "Greenzie's YouTube channel",
      "url": "https://www.youtube.com/c/Greenzie",
    },
  ],
  "mp3": Object {
    "sizeBytes": 225870038,
    "url": "https://ftp.osuosl.org/pub/ros/download.ros.org/sensethinkact/episodes/STA%20Ep%2011%20-%20CBQ.mp3",
  },
  "mp3SizeBytes": 225870038,
  "number": 11,
  "slug": "11-cbq",
  "tags": Array [
    "startup",
    "software",
    "sensor fusion",
    "autonomous driving",
    "mowers",
    "safety",
    "dogfooding",
  ],
  "title": "Commercial-grade Autonomous Mowers, Safety, and Dogfooding",
  "url": "https://sensethinkact.com/episodes/11-cbq",
  "youtube": Object {
    "clips": Array [
      Object {
        "baseTitle": "Does it Detect Sprinkler Heads?",
        "videoId": "ahUd_2Uydps",
      },
      Object {
        "baseTitle": "Making Perfect Turns",
        "videoId": "odAvY-ojVAA",
      },
      Object {
        "baseTitle": "Seasonal Workers",
        "videoId": "6CfAz0z07I4",
      },
      Object {
        "baseTitle": "Leap of Faith",
        "videoId": "8oVBa43dQGw",
      },
      Object {
        "baseTitle": "Differences Between Kids and Poles",
        "videoId": "HrraVxtyxok",
      },
    ],
    "mainContent": Object {
      "baseTitle": "Commercial-grade Autonomous Mowers, Safety, and Dogfooding",
      "videoId": "pBWLSmoBj54",
    },
  },
}
`;

exports[`getPost should get an episode with a custom slug and mp3 URL 1`] = `
Object {
  "customMp3FileName": "STA Ep 0 - Welcome",
  "customSlug": "0-welcome",
  "description": "Audrow Nash introduces the Sense Think Act Podcast.",
  "duration": Object {
    "hours": 0,
    "minutes": 4,
    "seconds": 39,
  },
  "excerpt": "Audrow Nash introduces the Sense Think Act Podcast.",
  "mp3": Object {
    "sizeBytes": 9097894,
    "url": "https://ftp.osuosl.org/pub/ros/download.ros.org/sensethinkact/episodes/STA%20Ep%200%20-%20Welcome.mp3",
  },
  "mp3SizeBytes": 9097894,
  "number": undefined,
  "slug": "0-welcome",
  "tags": Array [
    "announcement",
  ],
  "title": "Welcome to Sense Think Act",
  "url": "https://sensethinkact.com/episodes/0-welcome",
  "youtube": Object {
    "mainContent": Object {
      "baseTitle": "Welcome to Sense Think Act",
      "videoId": "7PdIbR8lNGI",
    },
  },
}
`;
